bugbug/bug_features.py

```diff
@@ -257,6 +257,16 @@ def __call__(self, bug, **kwargs):
         return sum(1 for commit in bug['commits'] if commit['ever_backedout'])
 
 
+class components_touched(object):
+    def __call__(self, bug, **kwargs):
+        return list(set([ct for commit in bug['commits'] for ct in commit['components']]))
```

Comment:
You are creating a list, then turning it into a set. You can just create a set directly: `set(ct for commit in bug['commits'] for ...`.

Also, use `component` instead of `ct`, as `ct` could be confused with commit.